homeassistant/components/synology_dsm/__init__.py

```diff
@@ -206,6 +213,47 @@ def _async_migrator(entity_entry: entity_registry.RegistryEntry):
             entry, data={**entry.data, CONF_MAC: network.macs}
         )
 
+    # setup DataUpdateCoordinator
+    async def async_coordinator_update_data_surveillance_station():
+        """Fetch all surveillance station data from api."""
+        surveillance_station = api.surveillance_station
+        try:
+            async with async_timeout.timeout(10):
+                await hass.async_add_executor_job(surveillance_station.update)
+        except SynologyDSMAPIErrorException as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+        data = {
+            "cameras": {},
+            "home_mode": {
+                "info": {},
+                "state": False,
+            },
+        }
+        if SynoSurveillanceStation.CAMERA_API_KEY in api.dsm.apis:
+            for camera in surveillance_station.get_all_cameras():
+                data["cameras"][camera.id] = camera
+
+        if SynoSurveillanceStation.INFO_API_KEY in api.dsm.apis:
+            data["home_mode"]["info"] = await hass.async_add_executor_job(
+                surveillance_station.get_info
+            )
```

Comment:
Can we avoid fetching this every time?  There are a lot of api hits going on every 30 seconds.